llvm-project/stage1.cmake

```diff
@@ -1,45 +1,134 @@
+
+# Stage1 clang should default to lld rather than BFD.
+# TODO: it seems this was ignored by the runtimes build...
+# otherwise why would CMAKE_EXE_LINKER_FLAGS be necessary?
 set(CLANG_DEFAULT_LINKER "lld" CACHE STRING "")
+# Stage1 clang should default to compiler-rt rather than libgcc.
 # TODO: it seems this was ignored by the runtimes build...
 set(CLANG_DEFAULT_RTLIB "compiler-rt" CACHE STRING "")
+
+# Disable arc migrate. We don't use that, ever.
 set(CLANG_ENABLE_ARCMT OFF CACHE BOOL "")
+
+# Disable static analyzer. Don't need it for stage1.
 set(CLANG_ENABLE_STATIC_ANALYZER OFF CACHE BOOL "")
+
+# Disable plugin support. Don't need it, ever.
 set(CLANG_PLUGIN_SUPPORT OFF CACHE BOOL "")
+
+# The compiler builtins are necessary.
 set(COMPILER_RT_BUILD_BUILTINS ON CACHE BOOL "")
+
+# GWP ASAN fails to build without libexecinfo-dev. Don't need it for stage1.
 set(COMPILER_RT_BUILD_GWP_ASAN OFF CACHE BOOL "")
+
+# Don't need libfuzzer, ever.
 set(COMPILER_RT_BUILD_LIBFUZZER OFF CACHE BOOL "")
+
+# Don't need memprof, ever.
 set(COMPILER_RT_BUILD_MEMPROF OFF CACHE BOOL "")
+
+# Don't need ORC, ever.
 set(COMPILER_RT_BUILD_ORC OFF CACHE BOOL "")
+
+# Disable profiling support. Not necessary for stage1.
 set(COMPILER_RT_BUILD_PROFILE OFF CACHE BOOL "")
+
+# Disable sanitizer support. Not necessary for stage1.
 set(COMPILER_RT_BUILD_SANITIZERS OFF CACHE BOOL "")
+
+# Don't need xray.
 set(COMPILER_RT_BUILD_XRAY OFF CACHE BOOL "")
+
+# Compiler-rt is meant as a replacement to libgcc.
 set(COMPILER_RT_HAS_GCC_S_LIB OFF CACHE BOOL "")
+
+# Who needs docs anyways? Can't RTFM if there is no FM.
 set(LIBUNWIND_INCLUDE_DOCS OFF CACHE BOOL "")
+
+# We don't plan to run the tests, disable them.
 set(LIBUNWIND_INCLUDE_TESTS OFF CACHE BOOL "")
+
+# We need the headers available when we install-libunwind.
 set(LIBUNWIND_INSTALL_HEADERS ON CACHE BOOL "")
+
+# Libunwind should use compiler-rt rather than libgcc.
 set(LIBUNWIND_USE_COMPILER_RT ON CACHE BOOL "")
+
+# TODO: I forget why I enabled this.
 set(LIBCXXABI_ENABLE_STATIC_UNWINDER ON CACHE BOOL "")
+
+# We don't plan to run the tests; don't build them.
 set(LIBCXXABI_INCLUDE_TESTS OFF CACHE BOOL "")
+
+# We want libc++abi to use compiler-rt.
 set(LIBCXXABI_USE_COMPILER_RT ON CACHE BOOL "")
+
+# We want libc++abi to use LLVM's libunwind.
 set(LIBCXXABI_USE_LLVM_UNWINDER ON CACHE BOOL "")
+
+# We want libc++ to use libc++abi.
 set(LIBCXX_CXX_ABI libcxxabi CACHE BOOL "")
+
+# We don't need C++ experimental APIs.
 set(LIBCXX_ENABLE_EXPERIMENTAL_LIBRARY OFF CACHE BOOL "")
+
+# We want libc++ to use compiler-rt. If we don't disable libatomic explicitly,
+# libc++ will try to link against it.
 set(LIBCXX_HAS_ATOMIC_LIB OFF CACHE BOOL "")
 set(LIBCXX_HAS_GCC_LIB OFF CACHE BOOL "")
 set(LIBCXX_HAS_GCC_S_LIB OFF CACHE BOOL "")
+
+# This is critical; libc++ depends on a libc. This MUST be set when building
+# against musl otherwise expect libc++ build failures related to locales.
```

Comment:
I think this is a slightly misleading comment.  The reason that we need to set this is because the default libc is assumed to be glibc, however we are building with musl.  The failures are related to the differences in assumptions.

Reply:
Well, it's not intentionally misleading. It's not malicious.

If this is not set, we will experience baffling failures related to locales.

Can you help me rephrase it?

Reply:
Sorry didn't mean to imply it was malicious, just that its not very precise.

How does something like this sound?
```
The C++ standard library requires the C library.  libc++ assumes that the default C library is glibc due to the prevalence, however, as we are using musl, we need to indicate that we are using musl to prevent failures due to incorrect assumptions. 
```